StreetHawk/Classes/Notification/Publish/SHApp+Notification.m

```diff
@@ -336,6 +339,41 @@ - (NSString *)apnsDeviceToken
     return [[NSUserDefaults standardUserDefaults] objectForKey:APNS_DEVICE_TOKEN];
 }
 
+- (void)setSHPushDenied
+{
+    if([self currentInstall].suid == nil){
+        return;
+    }
+
+    BOOL isPushPermissionDeniedBySysSettings = [[UIApplication sharedApplication] currentUserNotificationSettings].types == UIUserNotificationTypeNone;
+    BOOL isPushModuleInstalled = [[[NSUserDefaults standardUserDefaults] stringForKey:@"sh_module_push"]  isEqual: @"true"];
```

Comment:
This code is in StreetHawk/Classes/Notification/Publish/SHApp+Notification.m. 
sh_module_push is always true as long as you can reach this piece of code.